sbncode/CAFMaker/FillReco.cxx

```diff
@@ -61,10 +61,25 @@ namespace caf
   }
 
   void FillCRTHit(const sbn::crt::CRTHit &hit,
+                  uint64_t gate_start_timestamp,
                   bool use_ts0,
                   caf::SRCRTHit &srhit,
                   bool allowEmpty) {
-    srhit.time = (use_ts0 ? (float)hit.ts0_ns : hit.ts1_ns) / 1000.;
+
+    //srhit.time = (use_ts0 ? (float)hit.ts0_ns : hit.ts1_ns) / 1000.;
+
+    if(use_ts0){
+      if(gate_start_timestamp>0){
+        long long tmp_crttime = hit.ts0_ns-gate_start_timestamp; // ns
+        srhit.time = (float)tmp_crttime/1000.; // ns->us
+      }
+      else{
+        srhit.time = hit.ts0_ns/1000.-1600.;
```

Comment:
Where does 1600 come from? Is this worth a comment?

Reply:
Is this the time of the trigger in the MC? If so, wouldn't it be better to set that in an if branch...